File: jour_23/jour23.py
def cipher_cesaer(words, keys): 
    cipher_word = ""
    # En Python, on peut itérer directement sur les lettres !
    for char in words:
        if char.islower():
            # (Lettre - 'a' + clé) le tout modulo 26, puis on revient à 'a'
            nouveau_code = (ord(char) - ord('a') + keys) % 26 + ord('a')
            cipher_word += chr(nouveau_code)
        elif char.isupper(): 
            nouveau_code = (ord(char) - ord('A') + keys) % 26 + ord('A')
            cipher_word += chr(nouveau_code)
        else:
            # Si c'est un espace, un chiffre ou \n, on l'ajoute tel quel
            cipher_word += char
    return cipher_word
print("======= cipher cesaer ========")
word = input("Entrez le mot a chiffré : ")
while True :
    try :
        key = int(input(("Entrez la clé de chiffrement : ")))
        if key < 0 or key > 26 :
            print("la clé doit entre comprise entre 0 et 26\n")
        else :
            break
    except ValueError :
        print("Entrez une clé de chiffrement valide\n")
print(f'Le chiffrement César de "{word}" est : "{cipher_cesaer(word, key)}"')
input("Appuyez sur Entrée pour quitter...")

File: jour_23/jour23.pas
program CaesarSimple;

var
    word, cipher_word: string;
    key, i: integer;

begin
    writeln('======= Caesar cipher =======');
    write('Entrez le mot a chiffrer : ');
    readln(word);

    write('Entrez la clé de chiffrement (0-26) : ');
    readln(key);

    cipher_word := word;

    for i := 1 to length(word) do
    begin
        { Gestion des Majuscules }
        if (word[i] >= 'A') and (word[i] <= 'Z') then
            cipher_word[i] := chr((ord(word[i]) - ord('A') + key) mod 26 + ord('A'))
        
        { Gestion des Minuscules }
        else if (word[i] >= 'a') and (word[i] <= 'z') then
            cipher_word[i] := chr((ord(word[i]) - ord('a') + key) mod 26 + ord('a'))
            
        { Pour tout le reste (espaces, ponctuations, chiffres), on garde tel quel }
        else
            cipher_word[i] := word[i];
    end;

    writeln('Resultat : ', cipher_word);
    write('Appuyez sur Entrée pour quitter...');
    readln;
end.